llvm/lib/IR/Verifier.cpp

```diff
@@ -3598,14 +3598,8 @@ void Verifier::visitCallBase(CallBase &Call) {
     Check(Callee->getValueType() == FTy,
           "Intrinsic called with incompatible signature", Call);
 
-  // Disallow calls to functions with the amdgpu_cs_chain[_preserve] calling
-  // convention.
-  auto CC = Call.getCallingConv();
-  Check(CC != CallingConv::AMDGPU_CS_Chain &&
-            CC != CallingConv::AMDGPU_CS_ChainPreserve,
-        "Direct calls to amdgpu_cs_chain/amdgpu_cs_chain_preserve functions "
-        "not allowed. Please use the @llvm.amdgpu.cs.chain intrinsic instead.",
-        Call);
+  // Verify if the calling convention of the callee is callable.
+  Check(isCallableCC(Call.getCallingConv()), "Callee is not callable", Call);
```

Comment:
```suggestion
  Check(isCallableCC(Call.getCallingConv()), "calling convention is does not permit calls", Call);
```

Reply:
FWIW, the verifier isn't very consistent about whether the first letter of an error message should be capitalized.

Reply:
It's not great, but the verifier errors are less user facing than most things